spec/unit/pops/loaders/loaders_spec.rb

```diff
@@ -234,15 +234,15 @@ def compile_and_get_notifications(code)
     end
 
     it "an autoloaded type can reference an autoloaded type alias from another module when dependency is present in metadata.json" do
-      File.stubs(:read).with(user_metadata_path).returns user_metadata.merge('dependencies' => [ { 'name' => 'test-usee'} ]).to_pson
+      Puppet::FileSystem.stubs(:read).with(user_metadata_path, {:encoding => 'utf-8'}).returns user_metadata.merge('dependencies' => [ { 'name' => 'test-usee'} ]).to_pson
       expect(eval_and_collect_notices(<<-CODE, node)).to eq(['ok'])
         assert_type(User::WithUseeZero, [0])
         notice(ok)
       CODE
     end
 
     it "an autoloaded type can reference an autoloaded type alias from another module when other module has it declared in init.pp" do
```

Comment:
This seems to be failing because the stubbing is firing for other files e.g. `<puppet dir> spec/fixtures/unit/pops/loaders/loaders/dependent_modules_with_metadata/environment.conf`

This is because we used to stub File.read and call PuppetFileSystem in https://github.com/puppetlabs/puppet/blob/master/lib/puppet/settings.rb#L1194 .  But now that you're stubbing Puppet::Filesystem.read, it's not working properly.